ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/LogAppendingCommitStrategy.scala

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.validator
+
+import com.daml.ledger.participant.state.kvutils.DamlKvutils.{
+  DamlLogEntry,
+  DamlLogEntryId,
+  DamlStateKey,
+  DamlStateValue
+}
+import com.daml.ledger.participant.state.kvutils.Envelope
+import com.daml.ledger.participant.state.v1.ParticipantId
+
+import scala.collection.breakOut
+import scala.concurrent.{ExecutionContext, Future}
+
+class LogAppendingCommitStrategy[Index](
+    ledgerStateOperations: LedgerStateOperations[Index],
+    keySerializationStrategy: StateKeySerializationStrategy)(
+    implicit executionContext: ExecutionContext)
+    extends CommitStrategy[Index] {
+  override def commit(
+      participantId: ParticipantId,
+      correlationId: String,
+      entryId: DamlLogEntryId,
+      entry: DamlLogEntry,
+      inputState: Map[DamlStateKey, Option[DamlStateValue]],
+      outputState: Map[DamlStateKey, DamlStateValue]): Future[Index] = {
+    ledgerStateOperations.writeState(outputState.map {
+      case (key, value) =>
+        (keySerializationStrategy.serializeStateKey(key), Envelope.enclose(value))
+    }(breakOut))
```

Comment:
The future returned by `writeState` needs to be "join" or flatmapped with the future returned by `appendToLog`.

Reply:
Yes, changed accordingly.